packages/discord.js-utilities/src/lib/PaginatedMessage.ts

```diff
@@ -1,65 +1,162 @@
 import type { APIMessage, User, TextChannel, NewsChannel, Message, MessageReaction, ReactionCollector } from 'discord.js';
 
+/**
+ * This is a [[PaginatedMessage]], a utility to paginate messages (usually embeds).
+ * You must either use this class directly or extend it.
+ *
+ * [[PaginatedMessage]] uses actions, these are essentially reaction emojis, when triggered run the said action.
+ * You can utilize your own actions, or you can use the [[PaginatedMessage.defaultActions]].
+ * [[PaginatedMessage.defaultActions]] is also static so you can modify these directly.
+ *
+ * [[PaginatedMessage]] also uses pages, these are simply {@link https://discord.js.org/#/docs/main/stable/class/APIMessage APIMessages}.
+ *
+ * @example
+ * ```typescript
+ * const handler = new PaginatedMessage();
+ * ```
+ *
+ * @example
+ * ```typescript
+ * // To utilize actions you can use the IPaginatedMessageAction by implementing it into a class.
+ *
+ * class ForwardAction implements IPaginatedMessageAction {
+ *   public id = '▶️';
+ *
+ *   public run({ handler }) {
+ *     if (handler.index !== handler.pages.length - 1) ++handler.index;
+ *   }
+ * }
+ *
+ * // You can also give the object directly.
+ *
+ * const StopAction: IPaginatedMessageAction {
+ *   id: '⏹️',
+ *   disableResponseEdit: true,
+ *   run: ({ response, collector }) => {
+ *     await response!.reactions.removeAll();
+ *     collector!.stop();
+ *   }
+ * }```
+ *
+ */
 export class PaginatedMessage {
+	/**
+	 * The pages to be converted to [[PaginatedMessage.messages]]
+	 */
 	public pages: MessagePage[];
+
+	/**
+	 * The pages which were converted from [[PaginatedMessage.pages]]
+	 */
 	public messages: (APIMessage | null)[] = [];
 
+	/**
+	 * The actions which are to be used.
+	 */
 	public actions = new Map<string, IPaginatedMessageAction>();
 
+	/**
+	 * The handler's current page/message index.
+	 */
 	public index = 0;
 
+	/**
+	 * The amount of time to idle before the paginator is closed. Defaults to `20 * 1000`.
+	 */
 	public idle = 20 * 1000;
 
+	/**
+	 * Constructor for the [[PaginatedMessage]] class
+	 * @param __namedParameters The [[PaginatedMessageOptions]] for this instance of the [[PaginatedMessage]] class
```

Comment:
```suggestion
	 * @param options The [[PaginatedMessageOptions]] for this instance of the [[PaginatedMessage]] class
```

Reply:
Wrong. TypeDoc named destructured parameters as `__namedParameters`. Using `options` here will cause a mismatch.

### With `options`:
***There is no description for the options***
![image](https://user-images.githubusercontent.com/4019718/104376706-ad06ff80-5525-11eb-8091-d8ae44c45570.png)


### With `__namedParameters`:
***There is a description for the options***
![image](https://user-images.githubusercontent.com/4019718/104376771-c90aa100-5525-11eb-88a5-16e8410f5ab2.png)